content/2016.12.prototyping-an-app-with-kinto.md

```diff
@@ -0,0 +1,145 @@
+It'll also be used to automatically generate the CRUD forms for our data, straight into the admin.
+
+Following our data model, here's what the JSON schema should look like:
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "title": {
+      "type": "string",
+      "title": "Title",
+      "description": "Short title for an expense"
+    },
+    "comment": {
+      "type": "string",
+      "title": "Comment",
+      "description": "Provide details if needed..."
+    },
+    "tag": {
+      "type": "string",
+      "enum": ["fuel", "repair", "insurance", "maintenance", "equipment"],
+      "title": "Tag",
+      "description": "A tag for this expense"
+    },
+    "price": {
+      "type": "number",
+      "title": "Price",
+      "description": "The cost for this expense"
+    }
+  }
+}
+```
+
+Note the `enum` attribute on the tag which will be rendered in the forms as a select, instead of a simple input text field.
+
+Let's still not submit this new collection just yet, as we'll want to add a [UI schema].
+
+## Add a UI schema to improve the forms
+
+Adding a [UI schema] will improve the way the admin will display our data, and the forms to add/edit it.
+
+```json
+{
+  "ui:order": [
+    "title",
+    "comment",
+    "tag",
+    "price"
+  ],
+  "comment": {
+    "ui:widget": "textarea"
+  }
+}
+```
+
+The `ui:order` will be used to display the fields in the order we want (json objects attributes aren't ordered), and we also specify that we'd like a `textarea` for the comment field, instead of a single line input text field.
+
+Nearly done! Just before submitting this new collection, let's configure the columns to be displayed in the record list page.
+
+## Final touches
+
+By default, the admin will only display one column in the record list page. It would be much more convenient to display the `tag` and `price` on top of the `title`:
+
+![kinto-admin-records-list-columns]({filename}/images/kinto-admin-records-list-columns.png)
+
+## Profit!
+
+We now have everything we need for our proof of concept!
+
+List the current records (there's none yet):
+![kinto-admin-list-records]({filename}/images/kinto-admin-list-records.png)
+
+Add a new record:
+![kinto-admin-add-record]({filename}/images/kinto-admin-add-record.png)
+
+Display the list of (sortable) records once we added a few:
+![kinto-admin-list-records-many]({filename}/images/kinto-admin-list-records-many.png)
+
+As you can see, without a single line of code and very small effort, we have a functional, albeit very basic, proof of concept!
+
+While this is a nice and easy way to have a proof of concept, the admin's role is not meant to fulfill all your needs. For example, kinto has no way to do aggregation, and filtering can't be done through the admin. For those (and more), you'll want to create your own application.
```

Comment:
tbh: Maybe that something we want to add to the admin, let's create issues there. And formulate that it is not possible yet, but we would be glad to have some help adding the feature.